### PR TITLE
Feat: update to latest shared

### PR DESCRIPTION
**Pull request by kylebonnici**
## Checklist

- [ ] Ensure all user facing text is spell checked.
- [ ] Bump version in package.json
- [ ] Update changelog
- [ ] Write tests if needed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "pc-nrfconnect-dtm",
-    "version": "2.1.0",
```
Comment (boundlesscalm) [inline package.json:3]: Did we usually bump a major version for nrfutil?

**Review by datenreisender — APPROVED** [review]
Contrary to the description this not only updates to the latest shared but also updates `chart.js` and related dependencies. That change brings even more changes along but is still welcome. And there are also other welcome changes being done, e.g. replace the usage of `connect` from `react-redux`.

While already working on this, you could also move towards our folder structure, but I can understand if you do not want to spend even more effort on this. And I would prefer to do it in separate commits (or even PRs) anyhow. ;-) 

I am unable to review all changes but from what I saw I discovered no problems and trust you checked it well enough.

File: src/DeviceSelector.tsx [hunk 2]
```diff
@@ -4,107 +4,114 @@
...
 import {
     Device,
     DeviceSelector,
+    DeviceSetupConfig,
     getAppDir,
+    jprogDeviceSetup,
     logger,
-} from 'pc-nrfconnect-shared';
+} from '@nordicsemiconductor/pc-nrfconnect-shared';
+import { DeviceTraits } from '@nordicsemiconductor/pc-nrfconnect-shared/typings/generated/nrfutil';
```
Comment (boundlesscalm) [inline src/DeviceSelector.tsx:17]: Is this type not imported from shared properly?
You're importing it from the generated typings.

File: src/Receiver/ReceiverChartView.tsx [hunk 3]
```diff
@@ -33,10 +33,10 @@ const ChartView = () => {
     const lastReceived = useSelector(getLastReceived);
     const isRunning = useSelector(getIsRunning);
     const isInTransmitterMode = useSelector(getIsInTransmitterMode);
-    const [maxY, setMaxY] = useState(0);
+    const [maxY, setMaxY] = useState(10);
 
     useEffect(() => {
-        if (isRunning) setMaxY(10);
+        if (isRunning) setMaxY(0);
```
Comment (boundlesscalm) [inline src/Receiver/ReceiverChartView.tsx:39]: Why were these 2 values swapped?

File: src/WarningView.tsx [hunk 4]
```diff
@@ -43,15 +42,15 @@ const WarningView = () => {
                     </Alert>
                     {device && readbackProtection === 'protected' && (
                         <Alert variant="warning" label="">
-                            <div className="d-flex align-items-center flex-wrap readback-protection-warning">
+                            <div className="d-flex align-items-center readback-protection-warning flex-wrap">
                                 Unable to verify compatible firmware because the
                                 selected device has readback protection enabled.
-                                <Button
+                                <button
```
Comment (boundlesscalm) [inline src/WarningView.tsx:48]: Did you check whether the button keeps the same styling?
If I remember right, even custom button used to get the height, padding and some default styling (though I may be wrong).

File: src/Receiver/ReceiverChartView.tsx (hunk 3 above)
Comment (kylebonnici) [reply src/Receiver/ReceiverChartView.tsx:39]: Somehow the chart JS acts differently and actually both should be 0 as we relay on line 93 in the same file to set it to what ever the max y axis value is. Note this bar is only sed to show a background to the actual data bar. so I needs to have the same value as max (y-axis) of the received packets

File: src/WarningView.tsx (hunk 4 above)
Comment (kylebonnici) [reply src/WarningView.tsx:48]: yes :)

**Pull request merged**